pf/tests/repr_test.go

```diff
@@ -0,0 +1,211 @@
+		name     string
+		provider provider.Provider
+		expect   autogold.Value
+	}
+
+	testCases := []testCase{
+		{
+			"single-nested-block",
+			&pb.Provider{
+				AllResources: []pb.Resource{{
+					ResourceSchema: schema.Schema{
+						Blocks: map[string]schema.Block{
+							"single_nested_block": schema.SingleNestedBlock{
+								Attributes: map[string]schema.Attribute{
+									"a1": schema.Float64Attribute{
+										Optional: true,
+									},
+								},
+							},
+						},
+					},
+				}},
+			},
+			autogold.Expect(`{
+  "resources": {
+    "_": {
+      "single_nested_block": {
+        "element": {
+          "resource": {
+            "a1": {
+              "optional": true,
+              "type": 3
+            }
+          }
+        },
+        "optional": true,
+        "type": 6
+      }
+    }
+  }
+}`),
+		},
+		{
+			"list-nested-block",
+			&pb.Provider{
+				AllResources: []pb.Resource{{
+					ResourceSchema: schema.Schema{
+						Blocks: map[string]schema.Block{
+							"list_nested_block": schema.ListNestedBlock{
+								NestedObject: schema.NestedBlockObject{
+									Attributes: map[string]schema.Attribute{
+										"a1": schema.Float64Attribute{
+											Optional: true,
+										},
+									},
+								},
+							},
+						},
+					},
+				}},
+			},
+			autogold.Expect(`{
+  "resources": {
+    "_": {
+      "list_nested_block": {
+        "element": {
+          "resource": {
+            "a1": {
+              "optional": true,
+              "type": 3
+            }
+          }
+        },
+        "optional": true,
+        "type": 5
+      }
+    }
+  }
+}`),
+		},
+		{
+			"map-nested-attribute",
+			&pb.Provider{
+				AllResources: []pb.Resource{{
+					ResourceSchema: schema.Schema{
+						Attributes: map[string]schema.Attribute{
+							"map_nested_attribute": schema.MapNestedAttribute{
+								NestedObject: schema.NestedAttributeObject{
+									Attributes: map[string]schema.Attribute{
+										"a1": schema.StringAttribute{
+											Optional: true,
+										},
+									},
+								},
+							},
+						},
+					},
+				}},
+			},
+			autogold.Expect(`{
+  "resources": {
+    "_": {
+      "map_nested_attribute": {
+        "element": {
+          "schema": {
+            "element": {
+              "resource": {
+                "a1": {
+                  "optional": true,
+                  "type": 4
+                }
+              }
+            },
+            "type": 6
+          }
+        },
+        "type": 6
+      }
+    }
+  }
+}`),
+		},
+		{
+			"object-attribute",
+			&pb.Provider{
+				AllResources: []pb.Resource{{
+					ResourceSchema: schema.Schema{
+						Attributes: map[string]schema.Attribute{
+							"object_attribute": schema.ObjectAttribute{
+								AttributeTypes: map[string]attr.Type{
+									"a1": types.StringType,
+								},
+							},
+						},
+					},
+				}},
+			},
+			autogold.Expect(`{
+  "resources": {
+    "_": {
+      "object_attribute": {
```

Comment:
Interestingly enough this may look like a single-nested block to shim.Schema; it doesn't carry the distinction is it an attr or block in shim.Schema. May be allright for now but can cause some trouble in the corner cases perhaps.